pandas/core/internals/construction.py

```diff
@@ -291,6 +293,15 @@ def ndarray_to_mgr(
         if values.ndim == 1:
             values = values.reshape(-1, 1)
 
+    elif (
+        using_copy_on_write()
+        and isinstance(values, np.ndarray)
+        and (dtype is None or is_dtype_equal(values.dtype, dtype))
+        and copy_on_sanitize
+    ):
+        values = np.array(values, copy=copy_on_sanitize)
+        values = _ensure_2d(values)
+
     elif isinstance(values, (np.ndarray, ExtensionArray, ABCSeries, Index)):
         # drop subclass info
         values = np.array(values, copy=copy_on_sanitize)
```

Comment:
Sorry, don't know if this changed from the original version or I missed it before, but what is this additional check exactly doing? Because it seems to do exactly the same as the next `elif` block, and whenever you fullfill the new elif check, you would also fulfill the existing one (since the existing elif will pass whenever `values` is an ndarray, but without the extra checks)

Reply:
Good point, this is a leftover from the order change, removed it